src/hooks/useListEventsCategories.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from "react"
+import { SessionEventAttributes } from "../entities/Event/types"
+import { EventCategoryAttributes } from "../entities/EventCategory/types"
+
+export default function useListEventsCategories(
+  events?: SessionEventAttributes[] | null,
+  categories?: EventCategoryAttributes[] | null
+) {
+  return useMemo(() => {
+    if (events && categories) {
+      let categoriesFound: string[] = []
+      events.map(
+        (item) => (categoriesFound = [...categoriesFound, ...item.categories])
+      )
+
+      const categoryList = new Set<string>()
```

Comment:
you can use

```suggestion
      const categoryList = new Set<string>(categoriesFound)
```

and remove the following loop

PD: remember this isn't C 😆